frontend/src/requests/request.js

```diff
@@ -0,0 +1,240 @@
+  'Access-Control-Allow-Origin': '*',
+};
+
+const request = {
+  create: async ({ entity, jsonData }) => {
+    try {
+      const response = await axios.post(entity, jsonData);
+      successHandler(response, {
+        notifyOnSuccess: true,
+        notifyOnFailed: true,
+      });
+      return response.data;
+    } catch (error) {
+      return errorHandler(error);
+    }
+  },
+  read: async ({ entity, id }) => {
+    try {
+      const response = await axios.get(entity + id);
+      successHandler(response, {
+        notifyOnSuccess: false,
+        notifyOnFailed: true,
+      });
+      return response.data;
+    } catch (error) {
+      return errorHandler(error);
+    }
+  },
+  update: async ({ entity, id, jsonData }) => {
+    try {
+      const response = await axios.put(entity + id, jsonData);
+      successHandler(response, {
+        notifyOnSuccess: true,
+        notifyOnFailed: true,
+      });
+      return response.data;
+    } catch (error) {
+      return errorHandler(error);
+    }
+  },
+  delete: async ({ entity, id }) => {
+    try {
+      const response = await axios.delete(entity + id);
+      successHandler(response, {
+        notifyOnSuccess: true,
+        notifyOnFailed: true,
+      });
+      return response.data;
+    } catch (error) {
+      return errorHandler(error);
+    }
+  },
+
+  get: async ({ entity }) => {
+    try {
+      const response = await axios.get(entity);
+      return response.data;
+    } catch (error) {
+      return errorHandler(error);
+    }
+  },
+  post: async ({ entity, jsonData }) => {
+    try {
+      const response = await axios.post(entity, jsonData);
+
+      return response.data;
+    } catch (error) {
+      return errorHandler(error);
+    }
+  },
+  put: async ({ entity, jsonData }) => {
+    try {
+      const response = await axios.patch(entity, jsonData);
+      successHandler(response, {
+        notifyOnSuccess: true,
+        notifyOnFailed: true,
+      });
+      return response.data;
+    } catch (error) {
+      return errorHandler(error);
+    }
+  },
+  patch: async ({ entity, jsonData }) => {
+    try {
+      const response = await axios.patch(entity, jsonData);
+      successHandler(response, {
+        notifyOnSuccess: true,
+        notifyOnFailed: true,
+      });
+      return response.data;
+    } catch (error) {
+      return errorHandler(error);
+    }
+  },
+
+  filter: async ({ entity, options = {} }) => {
+    try {
+      let filter = options.filter ? 'filter=' + options.filter : '';
+      let equal = options.equal ? '&equal=' + options.equal : '';
+      let query = `?${filter}${equal}`;
+
+      const response = await axios.get(entity + query);
+      successHandler(response, {
+        notifyOnSuccess: false,
+        notifyOnFailed: false,
+      });
+      return response.data;
+    } catch (error) {
+      return errorHandler(error);
+    }
+  },
+
+  search: async ({ entity, options = {} }) => {
```

Comment:
Em có thể code như sau
```suggestion
  search: async ({ entity, options = {} }) => {
  try {
    let query = Object.entries(options)
      .map(([key, value]) => `${key}=${value}`)
      .join('&');

    const response = await axios.get(`${entity}/search`, { params: options });

    successHandler(response);
    return response.data;
  } catch (error) {
    return errorHandler(error);
  }
},

```